rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -992,11 +994,114 @@ void getNextFederationCreationBlockHeight_afterRSKIP186AndNoValueInStorage_retur
         // Act
 
         FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
-        Optional<Long> actualValueOptional = federationStorageProvider.getNextFederationCreationBlockHeight(activations);
+        Optional<Long> actualValue = federationStorageProvider.getNextFederationCreationBlockHeight(activations);
+
+        // Assert
+
+        assertFalse(actualValue.isPresent());
+
+    }
+
+    @Test
+    void getLastRetiredFederationP2SHScript_beforeRSKIP186_storageIsNotAccessedAndReturnsEmpty() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(false);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        Script expectedScript = new P2shErpFederationBuilder().build().getP2SHScript();
+        byte[] serializedScript = BridgeSerializationUtils.serializeScript(expectedScript);
+        // Putting some value in the storage just to then assert that before fork, the storage won't be accessed.
+        storageAccessor.saveToRepository(LAST_RETIRED_FEDERATION_P2SH_SCRIPT_KEY.getKey(), serializedScript);
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Assert
+        assertEquals(Optional.empty(), federationStorageProvider.getLastRetiredFederationP2SHScript(activations));
+
+    }
+
+    @Test
+    void getLastRetiredFederationP2SHScript_afterRSKIP186_getsValueFromStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        Script expectedScript = new P2shErpFederationBuilder().build().getP2SHScript();
```

Comment:
```suggestion
        Script expectedScript = new P2shErpFederationBuilder().build().getDefaultP2SHScript();
```
The default redeem script is the one stored. So using that value makes the test more realistic